### PR TITLE
WIP: Support Scala 2.13.0-RC2

### PR DESCRIPTION
**Pull request by rossabaker**

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -49,20 +49,13 @@ lazy val commonSettings = Seq(
...
-    "org.typelevel" %%% "cats-core" % "2.0.0-M1",
-    "org.typelevel" %%% "cats-laws" % "2.0.0-M1" % "test",
-    "org.typelevel" %%% "cats-effect" % "2.0.0-M1",
-    "org.typelevel" %%% "cats-effect-laws" % "2.0.0-M1" % "test",
-    "org.scalatest" %%% "scalatest" % "3.0.8-RC2" % "test"
+    "org.typelevel" %%% "cats-core" % "2.0.0-M2",
+    "org.typelevel" %%% "cats-laws" % "2.0.0-M2" % "test",
+    "org.typelevel" %%% "cats-effect" % "2.0.0-M2",
+    "org.typelevel" %%% "cats-effect-laws" % "2.0.0-M2" % "test",
+    "org.scalacheck" %%% "scalacheck" % "1.14.0" % "test",
```
Comment (rossabaker) [inline build.sbt]: If we're on cats-laws-2.x and cats-effect-laws-2.x, we should be on scalacheck-1.14.0.

File: build.sbt [hunk 2]
```diff
@@ -73,7 +66,7 @@ lazy val commonSettings = Seq(
     implicit val contextShiftIO: ContextShift[IO] = IO.contextShift(global)
     implicit val timerIO: Timer[IO] = IO.timer(global)
   """,
-  doctestTestFramework := DoctestTestFramework.ScalaTest,
+  doctestTestFramework := DoctestTestFramework.ScalaCheck,
```
Comment (rossabaker) [inline build.sbt]: Working around https://github.com/tkawachi/sbt-doctest/issues/130

**Comment by rossabaker** [comment]
Blocked on https://github.com/scalatest/scalatestplus-testng/issues/3.  We should consider porting those tests to reduce test dependency liability.

**Comment by rossabaker** [comment]
Nope, easier said than done: it (appropriately) uses the reactive-streams-tck, which uses testng.

**Comment by rossabaker** [comment]
Also blocked on https://github.com/scodec/scodec-bits/pull/118.

**Comment by mpilquist** [comment]
I was hoping to get #1478 in first given the ScalaTest changes. WDYT?

**Comment by rossabaker** [comment]
That looks reasonable, and this is blocked on testng, so :+1:*2.

**Comment by mpilquist** [comment]
@rossabaker I updated this and fixed a few test issues introduced by the new test suite (unrelated to this PR).

**Comment by rossabaker** [comment]
What is the version strategy going to be for this?  fs2 doesn't publish laws (I don't think?), so it doesn't have to be major.  But cats-effect-2.x will not be binary compatible in 2.11, and if the `fromFuture` ticket goes through, anywhere.  Does this mean fs2-2.0?

**Comment by mpilquist** [comment]
Argh, good catch. Maybe 1.1?

**Comment by mpilquist** [comment]
I'm going to merge this to series/1.0 at least and then figure out where to go from there.

**Pull request merged**